### PR TITLE
[CELEBORN-1247] Output config's alternatives to doc

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
Add configs' alternatives to doc.

### Why are the changes needed?
To help users use correct configs.

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
GA.

### CODE REVIEW

File: common/src/test/scala/org/apache/celeborn/ConfigurationSuite.scala [hunk 1]
```diff
@@ -138,8 +139,8 @@ class ConfigurationSuite extends AnyFunSuite {
   }
 
   def appendConfigurationTableHeader(output: ArrayBuffer[String]): Unit = {
-    output += "| Key | Default | Description | Since |"
-    output += "| --- | ------- | ----------- | ----- |"
+    output += "| Key | Default | Description | Since | Alternatives |"
```
Comment (pan3793) [inline common/src/test/scala/org/apache/celeborn/ConfigurationSuite.scala:142]: should we tip that the "Alternatives" is deprecated and will be removed in the future?

Comment (FMX) [reply common/src/test/scala/org/apache/celeborn/ConfigurationSuite.scala:142]: Sounds reasonable.

Comment (SteNicholas) [reply common/src/test/scala/org/apache/celeborn/ConfigurationSuite.scala]: @FMX, IMO, it's recommended to introduce how to upgrade config from `Alternatives`  especially network which requires users to choose module.

Comment (FMX) [reply common/src/test/scala/org/apache/celeborn/ConfigurationSuite.scala]: OK, I'll update this RP.

Comment (FMX) [reply common/src/test/scala/org/apache/celeborn/ConfigurationSuite.scala]: @SteNicholas Updated. How about now?

**Review by SteNicholas — APPROVED** [review]

**Comment by FMX** [comment]
Merged into main(v0.5.0) and branch-0.4(v0.4.0).

**Comment by pan3793** [comment]
there are merge issue for branch-0.4, a followup hotfix is required.

**Pull request closed without merging**